src/components/battleComponets/DisplayLives.jsx

```diff
@@ -1,23 +1,46 @@
-//import { useState } from "react";
-//import { useNavigate } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { socket } from "../../socket.js"
 
 function DisplayLives(props) {
+    let [myLife, setMyLife] = useState(props.maxLives);
+    let [oppLife, setOppLife] = useState(props.maxLives);
+
+    useEffect(()=>{
+
+        function lifeUpdateFunc(data){
+            console.log("lives:", data)
+            setMyLife(data)
+        }
+        function lifeUpdateOppFunc(data){
+            console.log("lives:", data)
+            setOppLife(data)
+        }
+
+        socket.on("lifeUpdate", lifeUpdateFunc)
+        socket.on("lifeUpdateOpp", lifeUpdateOppFunc)
+ 
+        return () => {
+            socket.off("lifeUpdate", lifeUpdateFunc)
+            socket.off("lifeUpdateOpp", lifeUpdateOppFunc)
+        };
+    },[])
+
```

Comment:
Hvis jeg skulle have gjort det, ville jeg nok have haft serveren til at sende liv for begge personer under ét event, og derved kun have haft brug for at have en state der havde holdt begge, så sparer man lidt logik, men dette er fint :)